hasher-matcher-actioner/hmalib/lambdas/actions/action_evaluator.py

```diff
@@ -97,26 +102,35 @@ def lambda_handler(event, context):
                     )
                     config.sqs_client.send_message(
                         QueueUrl=config.reactions_queue_url,
-                        MessageBody=threat_exchange_reaction_message.to_aws_message(),
+                        MessageBody=threat_exchange_reaction_message.to_aws_json(),
                     )
 
     return {"evaluation_completed": "true"}
 
 
-def get_action_labels(
+def get_actions_to_take(
     match_message: MatchMessage, action_rules: t.List[ActionRule]
-) -> t.Set[ActionLabel]:
+) -> t.Dict[ActionLabel, t.List[ActionRule]]:
     """
     Returns action labels for each action rule that applies to a match message.
     """
     classifications_by_match = get_classifications_by_match(match_message)
-    action_labels: t.Set[ActionLabel] = set()
+    action_label_to_action_rules: t.Dict[ActionLabel, t.List[ActionRule]] = dict()
     for classifications in classifications_by_match:
         for action_rule in action_rules:
             if action_rule_applies_to_classifications(action_rule, classifications):
-                action_labels.add(action_rule.action_label)
-    action_labels = remove_superseded_actions(action_labels)
-    return action_labels
+                if action_rule.action_label in action_label_to_action_rules:
+                    action_label_to_action_rules[action_rule.action_label].append(
+                        action_rule
+                    )
+                else:
+                    action_label_to_action_rules[action_rule.action_label] = [
+                        action_rule
+                    ]
+    action_label_to_action_rules = remove_superseded_actions(
+        action_label_to_action_rules
+    )
```

Comment:
Where did we land on adding classifications onto the match itself? What was davids perspective?

Reply:
Had a talk with @Dcallies and we'll move applying classification labels to be a concern of the matcher in a future PR.